app/src/main/java/org/oppia/android/app/drawer/ExitProfileDialogFragment.kt

```diff
@@ -52,19 +86,42 @@ class ExitProfileDialogFragment : DialogFragment() {
       .Builder(ContextThemeWrapper(activity as Context, R.style.AlertDialogTheme))
       .setMessage(R.string.home_activity_back_dialog_message)
       .setNegativeButton(R.string.home_activity_back_dialog_cancel) { dialog, _ ->
-        if (isFromNavigationDrawer) {
-          exitProfileDialogInterface.markHomeMenuCloseDrawer()
+        if (restoreLastCheckedMenuItem) {
+          exitProfileDialogInterface.checkLastCheckedItemAndCloseDrawer(
+            when (argument.number) {
+              2 -> Argument.IsAdministratorControlsSelected(
+                exitProfileDialogArguments.isAdministratorControlsSelected
+              )
+              else -> Argument.LastCheckedMenuItem(
+                getNavigationDrawerItem(
+                  exitProfileDialogArguments.lastCheckedMenuItemValue
```

Comment:
Should we create a constant with value - 2, so that it looks cleaner and expressive?
Or do we have a better way to handle it?